### ISSUE TITLE
Crashlytics NDK Proguard config is incorrect

### ISSUE THREAD

**Issue by nathan3d** [firebase/firebase-android-sdk#6593]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Jellyfish
  * Firebase Component: Crashlytics NDK
  * Component version: 19.2.1

### [REQUIRED] Step 3: Describe the problem

The keep rules for the Crashlytics NDK are limited to public classes, but at the minimum JniNativeApi is not a public class, so the included rules don't actually apply.

The Android Gradle Plugin includes this rule that mitigates the issue when used, but for builds using other build systems, this may not be present. 
```
-keepclasseswithmembernames,includedescriptorclasses class * {
    native <methods>;
}
```

**Comment by lehcar09** [comment]
Hi @nathan3d, thank you for reaching out. According to Firebase SDK [docs](https://github.com/firebase/firebase-android-sdk?tab=readme-ov-file#proguarding), 

> Firebase SDKs do not proguard themselves, but support proguarding. Firebase SDKs themselves are proguard friendly, but the dependencies of Firebase SDKs may not be.

From what I gather, Android Gradle Plugin (AGP) seamlessly integrates ProGuard into the build process; other build systems require manual configuration. For more information, you can check this for [reference](https://www.guardsquare.com/manual/setup/standalone).

However, based on proguard [documentation](https://www.guardsquare.com/manual/configuration/examples#:~:text=Processing%20callback%20methods,-If%20your%20application&This%20will%20preserve%20the%20given,from%20being%20removed%20or%20renamed.) for processing native methods,

> ProGuard doesn't look at your native code, so it won't automatically preserve the classes or class members that are invoked by the native code.

I suggest checking [DexGuard](https://www.guardsquare.com/dexguard), this is a commercial extension of ProGuard. DexGuard is specialized for Android applications and libraries: it optimizes and obfuscates not just the bytecode, but also the manifest file, resources, resource files, asset files, and native libraries.

**Comment by nathan3d** [comment]
Hi @lehcar09,

There are Proguard/R8 rules for the NDK published as part of the project here: https://github.com/firebase/firebase-android-sdk/blob/main/firebase-crashlytics-ndk/firebase-crashlytics-ndk-proguard.txt.

They're just not correct.

**Comment by nathan3d** [comment]
Right. The problem is JNINativeApi isn't public, so this rule has no effect.
Replacing:
`
-keep public class com.google.firebase.crashlytics.ndk.JniNativeApi { native <methods>; }
`
with:
`
-keep class com.google.firebase.crashlytics.ndk.JniNativeApi { native <methods>; }
`
would fix the issue, at least for `JniNativeApi`. I think some of the others may need updating as well. I haven't had a chance to verify all of the rules.

**Comment by lehcar09** [comment]
Hey @nathan3d, were currently working on this. Unfortunately, we are having trouble in reproducing the issue. By any chance, can you share an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) to help us investigate the issue? Thanks!
